internal/surgeon/surgeon.go

```diff
@@ -16,19 +17,105 @@ func CopyPage(path string, srcPage common.Pgid, target common.Pgid) error {
 }
 
 func ClearPage(path string, pgId common.Pgid) error {
+	return ClearPageElements(path, pgId, 0, -1)
+}
+
+func ClearPageElements(path string, pgId common.Pgid, start, end int) error {
 	// Read the page
 	p, buf, err := guts_cli.ReadPage(path, uint64(pgId))
 	if err != nil {
 		return fmt.Errorf("ReadPage failed: %w", err)
 	}
 
-	// Update and rewrite the page
-	p.SetCount(0)
-	p.SetOverflow(0)
+	if !p.IsLeafPage() && !p.IsBranchPage() {
+		return fmt.Errorf("can't clear elements in %q page", p.Typ())
+	}
+
+	elementCnt := int(p.Count())
+
+	if elementCnt == 0 {
+		return nil
+	}
+
+	if start < 0 || start >= elementCnt {
+		return fmt.Errorf("the start index (%d) is out of range [0, %d)", start, elementCnt)
+	}
+
+	if (end < 0 || end > elementCnt) && end != -1 {
+		return fmt.Errorf("the end index (%d) is out of range [0, %d]", end, elementCnt)
+	}
+
+	if start > end && end != -1 {
+		return fmt.Errorf("the start index (%d) is bigger than the end index (%d)", start, end)
+	}
+
+	if start == end {
+		return fmt.Errorf("invalid: the start index (%d) is equal to the end index (%d)", start, end)
+	}
+
+	preOverflow := p.Overflow()
+
+	if end == int(p.Count()) || end == -1 {
+		p.SetCount(uint16(start))
+		p.SetOverflow(0)
+		if preOverflow != 0 || p.IsBranchPage() {
+			if err := clearFreelist(path); err != nil {
```

Comment:
We have to do something on the freelist, otherwise some pages are neither reachable nor in the freelist, eventually it can't pass the `bbolt check`.
- either cleanup the freelist, so bbolt will scan the whole db to read all the freelist again. The bad side is reduce of performance on next startup. The good side is the implementation is simple. 
- Or update its parent page, all the way to the root if needed. Obviously it complicates the implementation.


```
$ ./bbolt page ./db 36
Page ID:    36
Page Type:  branch
Total Size: 4096 bytes
Overflow pages: 0
Item Count: 30

"0000": <pgid=4>
"0067": <pgid=2>
"0134": <pgid=3>
"10001": <pgid=9>
"10066": <pgid=10>
"10131": <pgid=5>
"10196": <pgid=6>
"20061": <pgid=7>
"20126": <pgid=11>
"20191": <pgid=12>
"30056": <pgid=13>
"30121": <pgid=8>
"30186": <pgid=15>
"40051": <pgid=16>
"40116": <pgid=14>
"40181": <pgid=18>
"50046": <pgid=19>
"50111": <pgid=17>
"50176": <pgid=21>
"60041": <pgid=22>
"60106": <pgid=20>
"60171": <pgid=24>
"70036": <pgid=25>
"70101": <pgid=23>
"70166": <pgid=27>
"80031": <pgid=28>
"80096": <pgid=26>
"80161": <pgid=30>
"90026": <pgid=31>
"90091": <pgid=32>
$ ./bbolt surgery clear-page-elements ./db --output ./new.db --pageId 36 --from 2 --to 3
$ ./bbolt check ./new.db 
page 3: unreachable unfreed
1 errors found
invalid value
```

Reply:
> Or update its parent page, all the way to the root if needed. Obviously it complicates the implementation.
I didn't get how it would help. I assume we are loosing OVERFLOW
```
$ ./bbolt surgery clear-page-elements ./db --output ./new.db --pageId 36 --from 2 --to 3
WARNING: The clearing has abandoned 1 page that is not yet referenced from free list. 
Consider `./bbolt surgery abandon-freelist ...`.
All elements with indexes in [2, 3) in page 36 were cleared

$./bbolt surgery abandon-freelist ./new.db --output ./new.db2

$ ./bbolt check ./new.db2
check OK
```

Reply:
> Or update its parent page, all the way to the root if needed. Obviously it complicates the implementation.
> I didn't get how it would help. I assume we are loosing OVERFLOW

Sorry for the confusion. I wanted to say: `If it's clearing branch page items, then we need to traverse all sub pages referenced by the cleared items, so as to figure out the pages which should be added into the freelist.`

> 
> $ ./bbolt surgery clear-page-elements ./db --output ./new.db --pageId 36 --from 2 --to 3
> WARNING: The clearing has abandoned 1 page that is not yet referenced from free list. 
> Consider `./bbolt surgery abandon-freelist ...`.
> All elements with indexes in [2, 3) in page 36 were cleared

If I understand it correctly, you are suggesting two things:
1. Add one more command `bbolt surgery abandon-freelist`;
2. Users must manually & explicitly execute the `surgery abandon-freelist`; we shouldn't do it automatically for users? 

Generally I agree with this. Just with two minor comments:
1. The page number may not be 1, so warning message will be something like "WARNING: The clearing has abandoned some page(s) that is not yet referenced from free list.";
2. If the freelist isn't synced yet, then do not print the warning.

Reply:
Done.  Let me finish the `bbolt surgery abandon-freelist` in a separate following PR.

Clear a branch element,
```
$ ./bbolt surgery clear-page-elements ./local-kv.db --output new.db --pageId 34 --from-index 1 --to-index 2
WARNING: The clearing has abandoned some pages that are not yet referenced from free list.
Please consider executing `./bbolt surgery abandon-freelist ...`
All elements in [1, 2) in page 34 were cleared
```

clear a leaf element,
```
$ ./bbolt surgery clear-page-elements ./local-kv.db --output new1.db --pageId 37 --from-index 1 --to-index 2
All elements in [1, 2) in page 37 were cleared
```